frame/membership/src/lib.rs

```diff
@@ -24,78 +24,123 @@
 
-	/// Required origin for setting or resetting the prime member.
-	type PrimeOrigin: EnsureOrigin<Self::Origin>;
+		/// Required origin for adding and removing a member in a single action.
+		type SwapOrigin: EnsureOrigin<Self::Origin>;
 
-	/// The receiver of the signal for when the membership has been initialized. This happens pre-
-	/// genesis and will usually be the same as `MembershipChanged`. If you need to do something
-	/// different on initialization, then you can change this accordingly.
-	type MembershipInitialized: InitializeMembers<Self::AccountId>;
+		/// Required origin for resetting membership.
+		type ResetOrigin: EnsureOrigin<Self::Origin>;
 
-	/// The receiver of the signal for when the membership has changed.
-	type MembershipChanged: ChangeMembers<Self::AccountId>;
+		/// Required origin for setting or resetting the prime member.
+		type PrimeOrigin: EnsureOrigin<Self::Origin>;
 
-	/// The maximum number of members that this membership can have.
-	///
-	/// This is used for benchmarking. Re-run the benchmarks if this changes.
-	///
-	/// This is not enforced in the code; the membership size can exceed this limit.
-	type MaxMembers: Get<u32>;
+		/// The receiver of the signal for when the membership has been initialized. This happens pre-
+		/// genesis and will usually be the same as `MembershipChanged`. If you need to do something
+		/// different on initialization, then you can change this accordingly.
+		type MembershipInitialized: InitializeMembers<Self::AccountId>;
 
-	/// Weight information for extrinsics in this pallet.
-	type WeightInfo: WeightInfo;
-}
+		/// The receiver of the signal for when the membership has changed.
+		type MembershipChanged: ChangeMembers<Self::AccountId>;
 
-decl_storage! {
-	trait Store for Module<T: Config<I>, I: Instance=DefaultInstance> as Membership {
-		/// The current membership, stored as an ordered Vec.
-		Members get(fn members): Vec<T::AccountId>;
+		/// The maximum number of members that this membership can have.
+		///
+		/// This is used for benchmarking. Re-run the benchmarks if this changes.
+		///
+		/// This is not enforced in the code; the membership size can exceed this limit.
+		type MaxMembers: Get<u32>;
 
-		/// The current prime member, if one exists.
-		Prime get(fn prime): Option<T::AccountId>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
 	}
-	add_extra_genesis {
-		config(members): Vec<T::AccountId>;
-		config(phantom): sp_std::marker::PhantomData<I>;
-		build(|config: &Self| {
-			let mut members = config.members.clone();
+
+	/// The current membership, stored as an ordered Vec.
+	#[pallet::storage]
+	#[pallet::getter(fn members)]
+	pub type Members<T: Config<I>, I: 'static = ()>	= StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+	/// The current prime member, if one exists.
+	#[pallet::storage]
+	#[pallet::getter(fn prime)]
+	pub type Prime<T: Config<I>, I: 'static = ()> = StorageValue<_, T::AccountId, OptionQuery>;
```

Comment:
```suggestion
	pub type Prime<T: Config<I>, I: 'static = ()> = StorageValue<_, T::AccountId>;
```

A storage item is `OptionQuery` by default, no need to specify it.

Reply:
I know it, but I like to specify it explicitly